keps/sig-node/20181106-in-place-update-of-pod-resources.md

```diff
@@ -132,48 +133,97 @@ Other identified non-goals are:
+(or maybe a followup in alpha), a new subresource, /resize, will be defined.
+This subresource could apply to other resources that carry PodTemplates, such
+as Deployments, ReplicaSets, Jobs, and StatefulSets.  Thie will allow users to
+grant RBAC access to controllers like VPA without allowing full write access to
+pod specs.
+
+The exact API here is TBD.
 
 #### Container Resize Policy
 
 To provide fine-grained user control, PodSpec.Containers is extended with
-ResizePolicy - a list of named subobjects (new object) that supports 'cpu'
-and 'memory' as names. It supports the following policy values:
-* NoRestart - the default value; resize Container without restarting it,
-* RestartContainer - restart the Container in-place to apply new resource
-  values. (e.g. Java process needs to change its Xmx flag)
-
-By using ResizePolicy, user can mark Containers as safe (or unsafe) for
-in-place resource update. Kubelet uses it to determine the required action.
+ResizePolicy - a list of named subobjects (new object) that supports 'cpu' and
+'memory' as names. It supports the following policy values:
+* RestartNotRequired - default value; try to resize the Container without
+  restarting it, if possible.
+* Restart - the container requires a restart to apply new resource values.
+  (e.g.  Java process needs to change its Xmx flag) By using ResizePolicy, user
+  can mark Containers as safe (or unsafe) for in-place resource update. Kubelet
+  uses it to determine the required action.
+
+Note: `RestartNotRequired` does not guarantee that container won't be
+restarted. The runtime may choose to stop the container if it is unable to
+apply the new resources without doing so.
 
 Setting the flag to separately control CPU & memory is due to an observation
 that usually CPU can be added/removed without much problem whereas changes to
 available memory are more probable to require restarts.
 
-If more than one resource type with different policies are updated, then
-RestartContainer policy takes precedence over NoRestart policy.
+If more than one resource type with different policies are updated at the same
+time, then any `Restart` policy takes precedence over `RestartNotRequired` policies.
+
+If a container's RestartPolicy is `Never`, the ResizePolicy must be set to
+`RestartNotRequired` to pass validation.  That said, any in-place resize may
+result in the container being stopped *and not restarted*, if the system can
+not perform the resize in place.
+
+#### Resize Status
+
+In addition to the above, a new field `Pod.Status.ContainerStatuses[i].Resize`
+will be added.  This field indicates whether kubelet has accepted or rejected a
+proposed resize operation.  Any time the
+`Pod.Spec.Containers[i].Resources.Requests` field differs from the
+`Pod.Status.ContainerStatuses[i].Resources` field, this new field explains why.
+
+This field can be set to one of the following values:
+* Unknown - the proposed resize (in Spec...Resources) has not been accepted or
```

Comment:
Does 'Pending' sound more informative?

Reply:
I agree. Proposed or Requested sounds much more intuitive.

Reply:
vote Requested

Reply:
Will change to Proposed